### PR TITLE
update telemetry event endpoint

### PR DESCRIPTION
**Pull request by creative-dev-lab**
### Context
telemetry event endpoint update.

### What's updated/implemented
- added iso string format for the date
- added created_at to the params
- updated API method to PUT

### Consideration
It's currently returning `UTC` timezone, Not sure if we need to return `UTC` timezone or the current timezone.

### Release type
- [x] Patch
- [ ] Minor
- [ ] Major

### Peer review
- [x] The feature/changes have been peer reviewed
- [ ] Automated test coverage and correctness have been peer reviewed
- [ ] The test plan has been peer reviewed

### CODE REVIEW

**Comment by johnAirRobe** [comment]
@creative-dev-lab I believe we are wanting to return UTC as that's what the telemetry events are currently using.

**Comment by creative-dev-lab** [comment]
That sounds good then! It's now returning UTC

File: Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift [hunk 1]
```diff
@@ -50,25 +50,37 @@ extension AirRobeEndpoint {
...
-                "department": department,
-                "itemCount": itemCount
-            ]
+            "properties": properties,
+            "created_at": Date().iso8601withFractionalSeconds
         ]
         return AirRobeEndpoint(
-            method: .POST,
+            method: .PUT,
             path: "/telemetry_events",
```
Comment (johnAirRobe) [inline Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift:84]: The new path is `/v1` and the new domain is `https://events.airrobe.com`.

File: Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift [hunk 2]
```diff
@@ -50,25 +50,37 @@ extension AirRobeEndpoint {
...
-                "brand": brand,
-                "material": material,
-                "category": category,
-                "department": department,
-                "itemCount": itemCount
-            ]
+            "properties": properties,
+            "created_at": Date().iso8601withFractionalSeconds
         ]
         return AirRobeEndpoint(
```
Comment (johnAirRobe) [inline Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift:82]: I'm not sure how it works in Swift but in widgets we needed to explicitly add this header: `"Content-Type": "application/json"`. This is a requirement for the API Gateway configuration we have.

**Review by johnAirRobe — CHANGES_REQUESTED** [review]

File: Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift (hunk 1 above)
Comment (creative-dev-lab) [reply Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift:84]: I see. okay. let me make the change! 👍

File: Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift (hunk 2 above)
Comment (creative-dev-lab) [reply Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift:82]: I added `application/json` to all the headers of the API call actually. You can find it

File: Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift (hunk 1 above)
Comment (creative-dev-lab) [reply Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift:84]: Does this work for both sandbox and production mode? @johnAirRobe

Comment (creative-dev-lab) [reply Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift]: Updated! 👍

**Review by piusj — APPROVED** [review]
LGTM, after resolving John's comments

File: Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift (hunk 1 above)
Comment (creative-dev-lab) [reply Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift]: @johnAirRobe could you review this again and let me know if it requires further touches?

File: Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift [hunk 3]
```diff
@@ -76,13 +76,14 @@ extension AirRobeEndpoint {
...
-            method: .POST,
-            path: "/telemetry_events",
+            method: .PUT,
+            path: "/v1",
             requestBody: requestBody,
-            host: configuration?.mode == .production ? AirRobeHost.airRobeConnectorProduction.rawValue : AirRobeHost.airRobeConnectorSandbox.rawValue
+            host: telemetryEventHost
```
Comment (johnAirRobe) [inline Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift:86]: I forgot to let you know that we currently don't have a telemetry endpoint for staging. I will create one for staging but I guess it's not a priority as TI probably won't be updating the version of the SDK anytime soon.

Actually, just use this domain for staging: `http://stg.events.airdemo.link` save you updating it later on. This is what the domain name will be when I create it.

**Review by johnAirRobe — CHANGES_REQUESTED** [review]

File: Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift (hunk 3 above)
Comment (creative-dev-lab) [reply Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift:86]: @johnAirRobe I've updated `sandbox` endpoint as `http://events.stg.airdemo.link` but it gave me an error that the hostname could not be found.
Is this intended behavior or anything went wrong?

Comment (johnAirRobe) [reply Sources/AirRobeWidget/Service/Endpoints/AirRobeWidgetEndpoints.swift]: I have yet to set up the infra to support this yet but will do soon.

**Review by johnAirRobe — APPROVED** [review]

**Pull request merged**